bundles/org.openhab.binding.luxtronikheatpump/src/main/java/org/openhab/binding/luxtronikheatpump/internal/LuxtronikHeatpumpHandler.java

```diff
@@ -0,0 +1,344 @@
+                try {
+                    // validate the value is valid
+                    HeatpumpCoolingOperationMode.fromValue(value);
+                } catch (InvalidOperationModeException e) {
+                    logger.warn("Heatpump {} mode recevieved invalid value {}: {}", channel.getCommand(), value,
+                            e.getMessage());
+                    return;
+                }
+                break;
+            case CHANNEL_EINST_KUHL_ZEIT_EIN_AKT:
+            case CHANNEL_EINST_KUHL_ZEIT_AUS_AKT:
+                float hours = ((DecimalType) command).floatValue();
+                value = (int) (hours * 10);
+                break;
+
+            default:
+                logger.debug("Received unknown channel {}", channelId);
+                break;
+        }
+
+        if (param != null && value != null) {
+            if (sendParamToHeatpump(param, value)) {
+                logger.debug("Heat pump mode {} set to {}.", channel.getCommand(), value);
+            } else {
+                logger.warn("Failed setting heat pump mode {} to {}", channel.getCommand(), value);
+            }
+        } else {
+            logger.warn("No valid value given for Heatpump operation {}", channel.getCommand());
+        }
+    }
+
+    @Override
+    public void initialize() {
+        LuxtronikHeatpumpConfiguration config = getConfigAs(LuxtronikHeatpumpConfiguration.class);
+
+        if (!config.isValid()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "At least one mandatory configuration field is empty");
+            return;
+        }
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        synchronized (scheduledFutures) {
+            logger.debug("try to connect to the heat pump every {} seconds to update the status", RETRY_INTERVAL);
+            ScheduledFuture<?> future = scheduler.scheduleWithFixedDelay(this::internalInitialize, 0, RETRY_INTERVAL,
+                    TimeUnit.SECONDS);
+            scheduledFutures.add(future);
+        }
+    }
+
+    private void internalInitialize() {
+        LuxtronikHeatpumpConfiguration config = getConfigAs(LuxtronikHeatpumpConfiguration.class);
+
+        // connect to heatpump and check if values can be fetched
+        HeatpumpConnector connector = new HeatpumpConnector(config.ipAddress, config.port);
+
+        try {
+            connector.read();
+        } catch (IOException e) {
+            setStatusConnectionError();
+            return;
+        }
+
+        // stop trying to establish a connection for initializing the thing once it was established
+        stopJobs();
+
+        // When thing is initialized the first time or and update was triggered, set the available channels
+        if (thing.getProperties().isEmpty() || tiggerChannelUpdate) {
+            try {
+                updateChannels(connector);
+            } catch (Exception e) {
```

Comment:
What type of exception do you expect here? Can you specify the concrete type?

Reply:
It's been a while since I've added that. Guess it was to make sure initalizing the thing can't fail at that point and the job to update the channels is set up even if it fails to update the channels here for some reason.

Reply:
It seems the method doesn't throw any reasonable exception. If it throws an Error, there might be something wrong fundamentally. This case is handled by the framework by logging all unchecked und uncought exceptions.

Reply:
let's assume a network timeout happens here as the device or network is temporary unavailable. Not catching the exception would result in not setting up the periodic update task. Shouldn't the exception been caught and instead the task been set up using at `try { } finally {}` instead?

Reply:
If I see correctly, `updateChannels()` doesn't perform any network actions. It also doesn't declare any `throws` in its method signature.

Reply:
ah. you're right. It used to trigger some network actions, but I removed that as part of another refactoring. Will remove the try catch